src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IObjectCreationExpression.cs

```diff
@@ -1142,5 +1142,1786 @@ static void M(bool b)
+                    Left: 
+                      IInvalidOperation (OperationKind.Invalid, Type: ?, IsInvalid) (Syntax: 'P2')
+                        Children(0)
+                    Right: 
+                      ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 4, IsInvalid) (Syntax: '4')
+
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: C1, IsInvalid, IsImplicit) (Syntax: 'x = new C1  ... 3, P2 = 4 }')
+              Left: 
+                ILocalReferenceOperation: x (IsDeclaration: True) (OperationKind.LocalReference, Type: C1, IsInvalid, IsImplicit) (Syntax: 'x = new C1  ... 3, P2 = 4 }')
+              Right: 
+                IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: C1, IsInvalid, IsImplicit) (Syntax: 'new C1 { C2 ... 3, P2 = 4 }')
+
+            IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null, IsInvalid) (Syntax: '(')
+              Expression: 
+                IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid) (Syntax: '')
+                  Children(0)
+
+            IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null, IsInvalid) (Syntax: 'P1 = 3')
+              Expression: 
+                ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: ?, IsInvalid) (Syntax: 'P1 = 3')
+                  Left: 
+                    IInvalidOperation (OperationKind.Invalid, Type: ?, IsInvalid) (Syntax: 'P1')
+                      Children(0)
+                  Right: 
+                    ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 3, IsInvalid) (Syntax: '3')
+
+            IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null, IsInvalid) (Syntax: 'P2 = 4 ')
+              Expression: 
+                ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: ?, IsInvalid) (Syntax: 'P2 = 4')
+                  Left: 
+                    IInvalidOperation (OperationKind.Invalid, Type: ?, IsInvalid) (Syntax: 'P2')
+                      Children(0)
+                  Right: 
+                    ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 4) (Syntax: '4')
+
+        Next (Regular) Block[B5]
+            Leaving: {R1}
+}
+
+Block[B5] - Exit
+    Predecessors: [B4]
+    Statements (0)
+";
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void ObjectCreationFlow_13()
+        {
+            string source = @"
+class C1
+{
+    public C2 C2 { get; set; } = new C2();
+
+
+    public void M(bool b)
+    /*<bind>*/{
+        var x = new C1 { C2 = { C31 = { [1] = 1, [2] = 2 }, C32 = { [3] = 3, [4] = 4 } } };
```

Comment:
>[2] [](start = 49, length = 3)

It would also be interesting to cover indexing into an array returned by a property and by a field

Reply:
Also,  indexing into an object with an indexer returned by a property and by a field  

---
In reply to: [184542172](https://github.com/dotnet/roslyn/pull/26389#discussion_r184542172) [](ancestors = 184542172)

Reply:
> It would also be interesting to cover indexing into an array returned by a property and by a field 
I'll add a prototype comment to add support for this. For this PR, I'm just going to support fields, events, and properties. I'll add array support at a later time, likely when I add support for array initializers as well.

---
In reply to: [184542172](https://github.com/dotnet/roslyn/pull/26389#discussion_r184542172) [](ancestors = 184542172)

Reply:
> Also,  indexing into an object with an indexer returned by a property and by a field  
I'll have this, as well as nested objects that have initializers.

---
In reply to: [185285865](https://github.com/dotnet/roslyn/pull/26389#discussion_r185285865) [](ancestors = 185285865,184542172)